siftseek/endpoints/helpers/query_helpers.py

```diff
@@ -0,0 +1,97 @@
+"""
+query_helpers.py provides helper functions for querying the database in the SiftSeek
+application.
+
+Each helper supports a custom error code for future development.
+"""
+
+from typing import Type, Any, List, Optional, TypeAlias
+
+from flask import abort
+from sqlalchemy import and_
+
+from siftseek.models.db import db, Base
+from siftseek.models.seeker import Seeker
+from siftseek.models.job_application import JobApplication
+
+# TODO Set up TypeAlias for Base
+
+
+def get_model_by_pk_or_404(model: Type[Base], pk: Any) -> Optional[Base]:
+    """
+    Tries to return a model from a database using its primary key. If no
+    instance is found, aborts with a 404.
+
+    Args:
+        model (Type[Base]): The model being searched for.
+        pk (Any): The primary key value used to identify the model instance.
+        error_code (int, optional): The HTTP error code. Defaults to 404.
+
+    Returns:
+        Optional[Base]: The model searched for or None if none found.
+
+    Raises:
+        HTTPException: If no instances are found, aborts and raises 404.
+    """
+    instance = db.session.get(model, pk)
+    if not instance:
+        abort(404)
+    return instance
+
+
+def get_job_apps_by_seeker_id_or_404(seeker_id: int) -> Optional[list[Base]]:
```

Comment:
this seems very specific to the jobSeeker model. i'm not as familiar with sqllite in python. But in node.js you can do something like 
```
job_apps = JobApplication.find({ "seeker_id": seeker_id })
```
Just curious if there is a syntax like that in sql lite. Cause then you can avoid this abstraction layer. Unless you feel it's necessary.

Reply:
I think I addressed this in the earlier comment, but essentially this is a pattern in flask-sqlalchemy that I was trying to replicate to get around a warning, but I think now that I'm not getting that warning anymore, I can do something similar to what you're suggesting in node.js with Flask.